src/poetry/console/commands/install.py

```diff
@@ -79,12 +80,20 @@ def handle(self) -> int:
             self.poetry.config.get("experimental.new-installer", False)
         )
 
-        extras = []
-        for extra in self.option("extras"):
-            if " " in extra:
-                extras += [e.strip() for e in extra.split(" ")]
-            else:
-                extras.append(extra)
+        if self.option("extras") and self.option("all-extras"):
+            raise ValueError(
+                "You cannot specify explicit extras while installing all extras"
+            )
```

Comment:
Since this is a user level error, let's replace this with either a `self.line_error` with `<error>` formatting; or a `PoetrySimpleConsoleError`. The `ValueError` iirc raises a stack trace when encountered.